warehouse/forklift/legacy.py

```diff
@@ -569,7 +572,27 @@ def _is_valid_dist_file(filename, filetype):
                 }:
                     return False
 
-    if filename.endswith(".exe"):
+    tar_fn_match = _tar_filenames_re.search(filename)
+    if tar_fn_match:
+        # Ensure that this is a valid tar file, and that it contains PKG-INFO.
+        z_type = tar_fn_match.group("z_type") or ""
+        try:
+            with tarfile.open(filename, f"r:{z_type}") as tar:
+                # This decompresses the entire stream until we find the tar
+                # file member we want.  Easy to DoS attack our CPU. :(
+                bad_tar = True
+                member = tar.next()
+                while member:
+                    parts = os.path.split(member.name)
+                    if len(parts) == 2 and parts[1] == "PKG-INFO":
+                        bad_tar = False
+                        break
```

Comment:
Breaking out here is optimizing for performance at the cost of potentially not detecting truncated or corrupted tar files that fail to parse beyond this point.

I'll leave it to the project owners to decide which behavior they want.

Reply:
i removed the break, it would allow invalid tars which defeats the point of validation.  cpu consumption is going to happen here no matter what and can be forced by any upload regardless of it.